### ISSUE TITLE
filemaker: navigation from ceramics module to CM results in disappearing pottery tab

### ISSUE THREAD

**Issue by lbestock** [arch-kiosk/arch-kiosk-office#1901]
Testing navigation and switching between SU ceramics/warecode ceramics on my iPad in PVD I encountered unexpected behavior when going back from making a ceramic record to the CM for the ceramic in warecode. 

My steps were as follows:
Configured for warecode pottery by clicking the button as a test user.
Logged out and in as myself.
Navigated to a previously (a few minutes previously in a different login) created CM bulk ceramic record.
Clicked on the pottery tab.
Was told it was awaiting warecode record (this is what I expected, though only after thinking about it).
Clicked in the header on the little pot to go to warecode recording.
Created two lines of warecode record.
Clicked in the header to go back to the CM record.

This is when it went wrong. I went back to the CM record as expected, but the pottery tab itself was now gone.
I could get back to warecode and that was fine, nothing messed up that record. But the tab that I had definitely seen previously (and from which I navigated to warecode) no longer existed.

I also noted that quantity and weight on the CM bulk page did not update with the warecode counts and weights, but that is a separate ticket.

**Comment by lbestock** [comment]
Very weird things going on, possibly not all relevant since the case of switching between warecode and SU in a real scenario is impossible. But some matters.

I uploaded and synchronized and downloaded again to go through a cycle and see what things looked like then.

The configuration for the ceramics module has reverted to what it was previously, not synchronizing my change there.

I think I had better give you this in terms of actual identifiers so you can look at it if you want.
809C is a bulk record that I had made when configured for warecode pottery. It is the one with the disappearing tab that led to this ticket. It still has no tab, and it still has the warecode stuff I recorded last round.
810C is a bulk record that I had made when configured for SU pottery. When I view it now (configured for warecode) it has the quantity and weight derived from the preprocessing I did, but the pottery tab (visible) tells me it is waiting for warecode analysis. I think that's right, right? Because I'm in warecode not SU.

If I go make the warecode record for this CM then again, navigating back, the pottery preprocessing tab is gone. Here the quantity and weight derived from the original preprocessing is still there.

**Comment by urapadmin** [comment]
okay, there are only two things here that I would look into:
- why is the pottery tab there in the first place for warecode pottery?
- why does the configuration for warecode pottery not synchronize?

The latter baffles me in particular since there is no difference in whether you change anything in the configuration or the code does it (which it did when changing). So are you really sure you about this one?

**Comment by urapadmin** [comment]
> But the tab that I had definitely seen previously (and from which I navigated to warecode) no longer existed.
That does not sound right to me. The part "from which I navigated to warecode." is only possible if you were in fact seeing the ceramics preprocessing tab. That has a big button that leads you to the ceramics module. If Warecode is on instead you can only navigate using the smaller pot button in the header.

I have tried this and cannot really do it in a way that leads to any error. Can you try to reproduce it again and perhaps see more of a pattern or an additional step?

**Comment by urapadmin** [comment]
and then please show me a screenshot with the first tab visible in the moment when you are back at cm with the pottery tab missing

**Comment by urapadmin** [comment]
Okay, I found out why the setting of the pottery module is not synchronizing: it is indeed not supposed to. Interesting. not sure, why I made that choice. I guess I thought I don't want anybody to change this permanently. So I have to initialize the templates for the different projects with the correct module myself. Hm. I really must not forget that with URAP or you would be changing this every day because it would never synchronize. 

Hm. 

...

I think, it is safer to make this key synchronize. I'll patch it.

**Comment by lbestock** [comment]
Happens again.
I created a new locus.
I created a new bulk CM in that locus.
I entered the bulk record and it looks like this:
![IMG_0089](https://user-images.githubusercontent.com/49291909/198839919-b94eae70-1612-47a8-8fed-67091b02dd62.PNG)

I clicked on the pottery tab.
I got the "it's coming soon" warning as expected.
While in the pottery tab, I clicked on the little pot button in the header next to the bulk number, and was asked if I really wanted to start warecode for that CM.
I did, and I made one line of recording in the pottery processing.
While in the warecode pottery processing, I clicked on the footsteps next to the PVD# in order to go back to my bulk record.
When I got there, it looked like this:
![IMG_0089](https://user-images.githubusercontent.com/49291909/198840006-fb2545cc-c682-4ad9-bd19-3d4a6650595f.PNG)

**Comment by lbestock** [comment]
Having just caught that I was still in there as admin, I repeated my steps as myself and the exact same thing happened: pottery tab before I create a warecode record, no pottery tab afterwards.

**Comment by urapadmin** [comment]
I got it. The "pottery" tab is supposed to appear only if there actually is a record in warecode analysis for the current cm. And that check was turned the other way around: It would only appear if there isn't any. So after you created a record and came back it disappeared. 

### recording 13.7.1
- fixed

**Comment by urapadmin** [comment]
@lbestock What about this one? Is it okay?

**Comment by lbestock** [comment]
I am still in 13.7 on PVD? From a download from last night?

**Comment by lbestock** [comment]
Mm but 13.7.2 is released this morning so let me give it a try.

**Comment by lbestock** [comment]
It's okay.
